ci/docker/gpu-debug.yaml

```diff
@@ -0,0 +1,23 @@
+spack:
+  specs:
+    - dla-future@develop build_type=Debug +cuda +miniapps +ci-test ^openblas ^mpich
+  view: false
+  concretization: together
+
+  packages:
+    all:
+      target: [x86_64]
+      variants:
+        - 'build_type=Release'
+        - 'cxxstd=17'
+    blaspp:
+      variants:
+        - '~cuda'
+        - '~openmp'
+    hpx:
+      variants:
+        - 'build_type=Debug'
+        - 'max_cpu_count=128'
+    mpich:
+      variants:
+        - '~fortran'
```

Comment:
I'm not sure, but probably the `packages.yaml` section can be extracted in a separate yaml file and shared among all environments (see https://spack.readthedocs.io/en/latest/environments.html#included-configurations)

Indeed, this should act just as hint for the concretizer (e.g. see that this is the `gpu-debug` and it has `all:build_type=Release` and `packages:blaspp:variants:~cuda`).

Reply:
Not sure. In this case I have a single file and changes are easy to track (i.e. if a change is made to this file a new tag for the build image is generated.)
With a shared package any change to it will create a new tag for all the build images even if the change doesn't affect all the builds.

`all:build_type=Release`: I don't see any reason to have all the dependencies built in debug mode. Sources are not available, so what's the advantage?
`packages:blaspp:variants:~cuda`: We don't use the blaspp cuda part, so why bothering to compile it?